### ISSUE TITLE
SIOOBE in IndentAction.getTextBlockIndentationString

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-jdt/eclipse.jdt.ui#532]
Using I20230416-1800.

I've tried to manually indent the begin of the string block by typing `<tab>` before `"""` 
in the new test I'm adding in `org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/JavadocTest_18.java` for https://github.com/eclipse-jdt/eclipse.jdt.core/issues/795:

```
public void testInlineReturn() {
	if(this.complianceLevel < ClassFileConstants.JDK16) {
		return;
	}
	this.runConformTest(
		new String[] {
			"X.java",
	"""
			/** Sample */
			public class Sample {
				/** {@return 42} */
				public int sample() {
					return 42;
				}
			}
					""",
		}
	);
}
```

and got this error:

```
java.lang.StringIndexOutOfBoundsException: begin 3, end 2, length 2
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4602)
	at java.base/java.lang.String.substring(String.java:2705)
	at java.base/java.lang.String.substring(String.java:2678)
	at org.eclipse.jdt.internal.ui.actions.IndentAction.getTextBlockIndentationString(IndentAction.java:1016)
	at org.eclipse.jdt.internal.ui.actions.IndentAction.computeReplaceData(IndentAction.java:398)
	at org.eclipse.jdt.internal.ui.actions.IndentAction.indentLine(IndentAction.java:549)
	at org.eclipse.jdt.internal.ui.actions.IndentAction.lambda$0(IndentAction.java:185)
	at org.eclipse.jdt.internal.ui.actions.IndentAction.run(IndentAction.java:219)
	at org.eclipse.ui.texteditor.AbstractTextEditor$ActivationCodeTrigger.verifyKey(AbstractTextEditor.java:791)
	at org.eclipse.jface.text.TextViewer$VerifyKeyListenersManager.verifyKey(TextViewer.java:480)
	at org.eclipse.swt.custom.StyledTextListener.handleEvent(StyledTextListener.java:70)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5855)
```

To reproduce, add the snippet above to the test, place cursor at the begin of the line with starting `"""` and hit `<tab>` key.

**Comment by iloveeclipse** [comment]
Seem to be regression from https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/33 ?

**Comment by jjohnstn** [comment]
@iloveeclipse Will look at it.

**Comment by jjohnstn** [comment]
Fixed via #534
